src/pages/MainPage/components/DirItem.js

```diff
@@ -0,0 +1,97 @@
+import {
+  requestAddDir,
+  requestAddDoc,
+  requestDelItem,
+  requestEditDir,
+} from "../../../services/api.js";
+import { getItem, setItem } from "../../../services/storage.js";
+
+export default function DirItem({
+  $target,
+  dirName = "",
+  reRender,
+  id = null,
+}) {
+  this.isDirOpen = getItem(id, true);
+  this.state = dirName;
+
+  this.setState = async (nextState) => {
+    this.state = nextState;
+    this.render();
+    await requestEditDir(id, this.state);
+  };
+
+  const $dirItemWrapper = document.createElement("div");
+  $dirItemWrapper.className = "dirItemWrapper";
+
+  const $dirContentWrapper = document.createElement("div");
+  const $dirIcon = document.createElement("i");
+  const $dirTitle = document.createElement("span");
+  $dirTitle.contentEditable = id === null ? false : true;
+  $dirTitle.className = "dirTitle";
+  $dirContentWrapper.className = "dirContentWrapper";
+  $dirIcon.className = `dirIcon fa-regular ${
+    this.isDirOpen ? "fa-folder-open" : "fa-folder"
+  }`;
+
+  const $dirBtnWrapper = document.createElement("div");
+  const $dirAddBtn = document.createElement("i");
+  const $dirDelBtn = document.createElement("i");
+  const $docAddBtn = document.createElement("i");
+  $dirBtnWrapper.className = "dirBtnWrapper";
+  $dirBtnWrapper.style.display = this.isDirOpen ? "" : "none";
+  $dirAddBtn.className = "fa-solid fa-folder-plus";
+  $dirDelBtn.className = "fa-solid fa-folder-minus";
+  $docAddBtn.className = "fa-regular fa-file";
+
+  $dirContentWrapper.append($dirIcon, $dirTitle);
+  $dirBtnWrapper.append($dirAddBtn, id === null ? "" : $dirDelBtn, $docAddBtn);
+  $dirItemWrapper.append($dirContentWrapper, $dirBtnWrapper);
+  $target.appendChild($dirItemWrapper);
+
+  this.render = () => {
+    $dirTitle.textContent = this.state;
+  };
+
+  this.toggleDir = (event) => {
```

Comment:
내부에서만 컨트롤하는 핸들러이니 this에 의존되있지 않아도 될 것 같네요

```suggestion
  const handleToggleDir = (event) => {
```